### PR TITLE
feat: Friendly prerender message + remove UglifyJS from SSR build

### LINKED ISSUE #178: Prerender error and workarounds
Preact-cli comes with prerender which render your `/` route using server-side rendering technique. So if you run `npm start` successfully but `npm run build` produces some errors, this mean that the prerender is failing because of some code requires DOM to work.

Workarounds:

- Wrap your with this to make sure it won't execute when prerendering. 

```
if (typeof window !== "undefined") {
  // Your code here
}
```

- Turn off preredering by run `npm run build -- --no-prerender`

### PR DESCRIPTION
**Pull request by rkostrzewski**
This closes #178

screen shot 2017-07-05 at 16 52 48

### CODE REVIEW

**Comment by rkostrzewski** [comment]
Heh, previously I've had `See https://github.com/developit/preact-cli#prerender for further information.` but I was too lazy to write it 😆

**Comment by lukeed** [comment]
Looks great @rkostrzewski ! 😻 

Quick question though: Why was uglify removed on SSR? UglifyJS was mangling the maps, ergo making tracing impossible?

**Comment by rkostrzewski** [comment]
@lukeed that's one of the reasons there other being that chief said so 😆, right @developit ?

**Comment by lukeed** [comment]
@rkostrzewski Lol 😂  Just went thru that Slack. Looks like Chief said that _because_ of the source map issues!

**Comment by hassanbazzi** [comment]
I notice this suggests people should check for window. I always just use this:

https://www.npmjs.com/package/window-or-global

It also makes SSR work for a bunch of features that need global, and it doesn't break even if you run things like eventListeners on global.

**Comment by rkostrzewski** [comment]
@thangngoc89 I've added `Pre-rendering` part to the `README.md` let me know if this is what you had in mind and I'll be merging this

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 - Fully **automatic code splitting** for routes
 - Transparently code-split any component with an [`async!`] prefix
 - Auto-generated [Service Workers] for offline caching powered by [sw-precache]
-- [PRPL](https://developers.google.com/web/fundamentals/performance/prpl-pattern/) pattern support for efficient loading
```
Comment (kurtextrem) [inline README.md]: why remove the link? :)

**Comment by thangngoc89** [comment]
@rkostrzewski LGTM

**Pull request merged**